modules/stage-2.nix

```diff
@@ -4,8 +4,20 @@
 #        (This'll allow complex schemes like LVM)
 
 let
-  rootfs = config.fileSystems."/".device;
+  rootfs = config.fileSystems."/";
+  inherit (builtins) concatStringsSep elem length;
   inherit (lib) mkMerge mkOrder;
+
+  waitForRootfs =
+    lib.optionalString (elem "loop" rootfs.options) ''
+      while [ ! -f "${rootfs.device}" ]; do
+        echo "Waiting for the rootfs image at '${rootfs.device}'..."
+        sleep 5
```

Comment:
This delay could be shorter for a faster boot. To avoid spamming the console, we could print a . without a newline each on each loop rather than repeating the "Waiting" message.